pkg/sfu/downtrack.go

```diff
@@ -400,20 +405,22 @@ func (d *DownTrack) SetStreamAllocatorReportInterval(interval time.Duration) {
 	d.streamAllocatorLock.Unlock()
 
 	go func(generation int) {
-		<-ticker.C
+		for {
+			<-ticker.C
```

Comment:
the loop will deadlock here and leak the goroutine after the ticker is stopped. maybe something like
```go
// in DownTrack def
resetStreamAllocatorReport chan struct{}

// in SetStreamAllocatorReportInterval
d.ClearStreamAllocatorReportInterval()

if interval == 0 {
    return
}

ticker := time.NewTicker(interval)
defer ticker.Stop()

reset := make(chan struct{})

d. streamAllocatorLock.Lock()
resetPrev := d.resetStreamAllocatorReport
d.resetStreamAllocatorReport = reset
d. streamAllocatorLock.Unlock()

if resetPrev != nil {
    close(resetPrev)
}

go func() {
    for {
        select {
        case <-reset:
            return
        case <-ticker.C:
            // ...
        }
    }
}

// in ClearStreamAllocatorReportInterval
d. streamAllocatorLock.Lock()
reset := d.resetStreamAllocatorReport
d.resetStreamAllocatorReport = nil
d. streamAllocatorLock.Unlock()

if resetPrev != nil {
    close(reset)
}
```
or you could use a fuse and assign an unused fuse in the constructor... then you wouldn't need the nil checks... anyway there needs to be something to exit that loop when the `DownTrack` goes away

Reply:
Thank you Paul. Good catch on the `Stop` hanging the loop. I think it should be fine to completely remove the ticker and just use a `timer` as you suggested in the `Prober` case. And let the tick happen and goroutine exit with generation mismatch. Makes things simpler. Will make that change and let you know.

Reply:
Cleaned it up in this [commit](https://github.com/livekit/livekit/pull/1512/commits/453227359b9d9cc62d979d981cd12f9790c259ef). Thank you  🙇

Reply:
yep, looks good 👍